cli/src/commands/diff.rs

```diff
@@ -0,0 +1,526 @@
+use clap::Parser as ClapParser;
+use codec::Decode;
+use color_eyre::eyre::eyre;
+use frame_metadata::RuntimeMetadataPrefixed;
+use std::collections::HashMap;
+use std::path::PathBuf;
+use std::str::FromStr;
+
+use crate::utils::FileOrUrl;
+use color_eyre::owo_colors::OwoColorize;
+use jsonrpsee::client_transport::ws::Uri;
+use scale_info::form::PortableForm;
+use scale_info::Variant;
+use subxt_codegen::utils::MetadataVersion;
+use subxt_metadata::{
+    ConstantMetadata, Metadata, PalletMetadata, RuntimeApiMetadata, StorageEntryMetadata,
+    StorageEntryType,
+};
+
+/// Explore the differences between two nodes
+///
+/// # Example
+/// ```
+/// subxt diff ./artifacts/polkadot_metadata_small.scale ./artifacts/polkadot_metadata_tiny.scale
+/// subxt diff ./artifacts/polkadot_metadata_small.scale wss://rpc.polkadot.io:443
+/// ```
+#[derive(Debug, ClapParser)]
+pub struct Opts {
+    node1: String,
+    node2: String,
+    #[clap(long, short)]
+    version: Option<MetadataVersion>,
```

Comment:
I don't get why `version` is a CLI parameter, how would one insert `--version` here and why? 

Better to read that from the metadata itself?

Reply:
The version is which version of metadata is being requested when a url is given (eg are we downloading V14 metadatas or unstable-v15 metadatas or "whatever is latest stable"; something like that)